app/src/main/java/com/example/bigowlapp/utils/AuthenticatorByPhoneNumber.java

```diff
@@ -0,0 +1,66 @@
+package com.example.bigowlapp.utils;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Attendance;
+import com.example.bigowlapp.model.AuthByPhoneNumberFailure;
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.model.UserScheduleResponse;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.RepositoryFacade;
+import com.google.firebase.Timestamp;
+
+public class AuthenticatorByPhoneNumber {
+
+    private String deviceIdNumber;
+    private String devicePhoneNumber;
+    private String currentUserPhoneNumber;
+    private final Context context;
+    private final AuthRepository authRepository = new AuthRepository();
+
+    RepositoryFacade repositoryFacade = RepositoryFacade.getInstance();
+
+    public AuthenticatorByPhoneNumber(Context context) {
+        this.context = context;
+    }
+
+    @SuppressLint("MissingPermission")
```

Comment:
On functions where you suppressed "MissingPermission", I think you should actually handle the permissions not being granted with an if statement. Right now any code with this Lint suppression is simply assuming the permissions were granted.

If you 100% certain you have the permission though, than I do think it is ok, and I would say you should  have a comment stating such, once again, for all cases where you used this SuppressLint line.

Based on what is said here, if you don't have the permission, but it tries to use something, that needs it, the app will just crash: http://tools.android.com/tips/lint-checks

> This check scans through your code and libraries and looks at the APIs being
used, and checks this against the set of permissions required to access those
APIs. If the code using those APIs is called at runtime, then the program will
crash.


Once again, this would apply to all section of the code with this SuppressLint, but I'm just commenting on this one.

Reply:
Yes, the permission was asked at the home page after sign in an the user can not proceed after sign in if that permission is not granted

Reply:
@Landry333 Oh, that's cool then. I would just say to add a comment on this function then, so none of us get confused, and to make it clear that you did this intentionally.

Reply:
ok